🐛(tasks) fix redeploying apps that have no deployment config and release 4.1.2

sampaccoud opened the pull request:
## Purpose

The redis-sentinel app has no dc and works only by declaring an svc. The first deployment works perfectly, but redeployments fail because we endup expecting 0 pods running when there are in fact pods running (they have been started by the k8s operator and not by a dc created
by Arnold).

## Proposal

Instead of checking if the number of running pods is equal to the number of expected pods, we can check if the number of running pods is greater than the number of expected pods.

sampaccoud committed:
🐛(tasks) fix redeploying apps that have no deployment config

The redis-sentinel app has no dc and works only by declaring an svc.
The first deployment works perfectly, but redeployments fail because
we endup expecting 0 pods running when there are in fact pods running
(they have been started by the k8s operator and not by a dc created
by Arnold).

lunika commented on tasks/manage_app.yml:
```diff
@@ -84,7 +84,7 @@
       - "status.phase=Running"
   register: pods
-  until: pods.resources is defined and ( pods.resources | length ) == ( expected_running_pods | int )
+  until: pods.resources is defined and ( pods.resources | length ) >= ( expected_running_pods | int )
```
we could also add a `when` instruction testing if `deployments | length > 0`

```
when: deployments |  length > 0
```

jmaupetit replied:
Indeed, I think your solution is more reliable @lunika, but why not using:

```yaml
  when: expected_running_pods | int > 0
```

sampaccoud replied:
Ok I did this. Checking on `expected_running_pods` is easier to understand as we have this variable at hand here.

lunika replied:
> Indeed, I think your solution is more reliable @lunika, but why not using:
> 
> ```yaml
>   when: expected_running_pods | int > 0
> ```

Yes, you are a very clever man

![](https://media.giphy.com/media/7XuPYJXaF1CBAmbwQQ/giphy.gif)

jmaupetit reviewed (commented):
Don't forget the CHANGELOG!

sampaccoud committed:
🔖(patch) bump release to 4.1.2

Fixed

- Checking the number of running pods was hanging when redeploying an already
  running `redis-sentinel` app.

sampaccoud commented:
Added the 4.1.2 fix release to this PR.

jmaupetit reviewed (approved):
![](https://media0.giphy.com/media/bzE1WAm8BifiE/giphy.gif)

sampaccoud commented:
@jmaupetit worse GTM gif I got from you!
